File: frontend/app/lib/types/prompt.ts
export enum PromptStatus {
  None = 'None',
  Pending = 'Pending',
  Processing = 'Processing',
  Completed = 'Completed',
  Failed = 'Failed',
}

export interface Prompt {
  id: string;
  text: string;
  status: PromptStatus;
  errorMessage?: string | null;
  outputText?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreatePromptDto {
  text: string;
}

File: backend/LlmDashboard.Domain/Models/Prompt.cs
using LlmDashboard.Domain.Enums;

namespace LlmDashboard.Domain.Models;

public class Prompt
{
    public required Guid Id { get; set; }
    public required string Text { get; set; }
    public required PromptStatus Status { get; set; }
    public string? ErrorMessage { get; set; }
    public string? OutputText { get; set; }
    
    public required DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}